doc/cookbook/index.md

```diff
@@ -0,0 +1,148 @@
+
+\snippet doc/tutorial/sequence_file/sequence_file_snippets.cpp record_type2
+
+# Writing records
+
+The easiest way to write to a sequence file is to use the seqan3::sequence_file_output::push_back()
+or seqan3::sequence_file_output::emplace_back() member functions.
+These work similarly to how they work on an std::vector.
+
+\include test/snippet/io/sequence_file/sequence_file_output_record_wise_iteration.cpp
+
+# File conversion
+
+\snippet doc/tutorial/sequence_file/sequence_file_snippets.cpp file_conversion
+
+# Define a custom scoring scheme
+
+\snippet doc/tutorial/pairwise_alignment/configurations.cpp include_scoring_scheme
+
+\snippet doc/tutorial/pairwise_alignment/configurations.cpp scoring_scheme
+
+\attention SeqAn's alignment algorithm computes the maximal similarity score, thus the match score must be set to a
+positive value and the score for mismatch and gaps must be negative in order to maximize over the matching letters.
+
+# Calculate edit distance for a set of sequences
+
+This recipe can be used to calculate the edit distance for all six pairwise combinations. Here we only allow at most 7
+errors and filter all alignments with 6 or less errors.
+
+\include doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp
+
+# Searching for matches
+
+This recipe can be used to search for all occurrences of a substring and print the number of hits and the
+positions in an ascending ordering.
+
+\include doc/tutorial/search/search_solution2.cpp
+
+If you want to allow for errors in your query, you need to configure the approximate search with a seqan3::search_cfg::max_error or seqan3::search_cfg::max_error_rate object.
+
+To search for either 1 insertion or 1 deletion you can use the seqan3::search_cfg::max_error:
+
+\snippet doc/tutorial/search/search_small_snippets.cpp error_search
+
+# Reading the CIGAR information into an actual alignment
+
+In SeqAn, the conversion from a CIGAR string to an alignment (two aligned_sequences) is done automatically for you. You can access it by querying seqan3::field::alignment from the record:
+
+\snippet doc/tutorial/alignment_file/alignment_file_snippets.cpp alignments_without_ref
+
+# Combining sequence and alignment files
+
+This recipe can be used to:
+1. Read in a FASTA file with the reference and a SAM file with the alignment
+2. Filter the alignment records and only take those with a mapping quality >= 30.
+3. For the resulting alignments, print which read was mapped against the reference id and the number of seqan3::gap's
+   involved in the alignment (either in aligned reference or in read sequence).
+
+\snippet doc/tutorial/alignment_file/alignment_file_solution2.cpp solution
+
+# Map reads ans write output to SAM file
+For a full recipe on creating your own readmapper, see the very end of the tutorial \ref tutorial_read_mapper.
+
+\snippet doc/tutorial/read_mapper/read_mapper_step4.cpp solution
+
+# Argument parser
```

Comment:
This headline doesn't tell me what's happening in the snippet? Is like a basic argument parser example?

Reply:
It is, I changed it to "Constructing a basic argument parser"